rollup/internal/controller/relayer/l2_relayer.go

```diff
@@ -473,6 +491,12 @@ func (r *Layer2Relayer) ProcessPendingBatches() {
 		log.Info("Forcing submission of batches due to timeout", "batch index", batchesToSubmit[0].Batch.Index, "first block created at", oldestBlockTimestamp)
 	}
 
+	// Sanity checks before constructing the transaction
+	if err := r.sanityChecksBeforeConstructingTransaction(batchesToSubmit); err != nil {
+		log.Error("Sanity checks failed before constructing transaction", "batches count", len(batchesToSubmit), "first batch index", batchesToSubmit[0].Batch.Index, "last batch index", batchesToSubmit[len(batchesToSubmit)-1].Batch.Index, "err", err)
```

Comment:
do we need to add a metric when get the sanityCheck failed error.

Reply:
I think it's OK since there are many alerts about committing/finalizing delay.